Tryouts/Core/Abstractions/Modules/IModule.cs

```diff
@@ -0,0 +1,10 @@
+﻿namespace MorganStanley.ComposeUI.Tryouts.Core.Abstractions.Modules;
+
+public interface IModule
+{
+    string Name { get; }
+    IObservable<LifecycleEvent> LifecycleEvents { get; }
+    Task Initialize();
```

Comment:
When would initialization and "launching" two separate actions? What would these steps do?

Also my previous question was around who would implement this interface? From the current PR it seems this would be an internal interface, we're not expecting clients to implement it - if so, I'd suggest making it internal.

Reply:
1. I was (am) debating this as well, and currently it's not really useful, the idea was using initialize to prepare the launch, e.g. book ports in some internal mechanism for self-hosted webapps, while launch would only do the actual launching. It's not doing that yet though.
2. This is mainly an internal compose interface, although I'm thinking if supporting externally added module types is desirable

Reply:
Changed visibilites to internal on internal components